packages/appium/lib/extension/package-changed.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+
+import { mkdirp } from '@appium/support';
+import { isPackageChanged } from 'package-changed';
+import path from 'path';
+import { HASHFILE_RELATIVE_PATH } from '../constants';
+import _ from 'lodash';
+import log from '../logger';
+
+/**
+ * Determines if extensions have changed, and updates a hash the `package.json` in `appiumHome` if so.
+ *
+ * If they have, we need to sync them with the `extensions.yaml` manifest.
+ *
+ * _Warning: this makes a blocking call to `writeFileSync`._
+ * @param {string} appiumHome
+ * @returns {Promise<boolean>} `true` if `package.json` `appiumHome` changed
+ */
+export async function packageDidChange (appiumHome) {
+  if (!appiumHome || !_.isString(appiumHome)) {
+    throw new TypeError('appiumHome must be a string');
```

Comment:
such error might be confusing for a user if they see it.

Reply:
if a user calls `packageDidChange` they have other problems 😄  

this exception is really for a contributor to appium who may call the function with something weird.

I could also just remove the conditional/exception and trust people will mind the types.

Reply:
if this is intended to be used by devs only then I assume they would be fine with default exception they'd get if a wrong method is called on appiumHome instance. The docs clearly states the var is expected to be a string

Reply:
you're right, I'm not sure why I did that.